### PR TITLE
Adding logger for dev mode for cart.

### PR DESCRIPTION
**Pull request by joshirohit100**
Sometimes we need to debug whats in cart and I think this will help.  cc @sdelbosc

### CODE REVIEW

File: Model/CartManagement.php [hunk 1]
```diff
@@ -443,10 +468,30 @@ public function getCart($cartId)
...
+        // If development mode on.
+        if ($this->appMode->getMode() == 'developer') {
+            $data = $this->dataProcessor->buildOutputDataArray($cart, 'Acquia\CommerceManager\Api\Data\CartWithTotalsInterface');
+            if (is_array($data)) {
+                $this->logger->debug(json_encode($data));
+            }
+        }
+    }
+
+
```
Comment (sdelbosc) [inline Model/CartManagement.php]: Remove this extra line

Comment (joshirohit100) [reply Model/CartManagement.php]: PR Updated!

**Comment by malachy-mcconnell** [comment]
Is this still work in progress?

We could also add a configuration option "log cart object in developer mode", or more generally, 'turn on verbose logging in developer mode' so we can use this pattern in other places. The idea being if the app-mode is developer *and* the config option for verbose logging is set, then do generate this (and other) extra logging.

What do you think?

**Comment by joshirohit100** [comment]
Initially did the same 😂. Regarding WIP, I think yes still

**Comment by malachy-mcconnell** [comment]
My main concern is exposure of personally identifiable information. So I will ensure we document that if developer mode is on PII fails (GDPR esp.)

But let's get this merged for now and improve logging-options, documentation, and acceptance of risk options later. Please fix trivial merge conflict and then I will merge this request.

**Comment by joshirohit100** [comment]
PR updated! Conflict resolved!

**Pull request merged**